### ISSUE TITLE
firmware: iOS 16 layout issues with bootstrap components

### ISSUE THREAD

**Issue by MainServerSpace** [opnsense/core#5635]
The audit button dropdown does not work on mobile devices but can verify it works on Desktop. Any ideas?

**Comment by EmperorsLoungeVPN** [comment]
https://user-images.githubusercontent.com/86547540/159139009-67d1b468-14ea-47c1-b6be-1f72f21bce97.MOV

**Comment by EmperorsLoungeVPN** [comment]
Physically have to close out the side menu for it to go to status page also.

**Comment by MainServerSpace** [comment]
Any update on a fix

**Comment by MainServerSpace** [comment]
Would be nice to see the mobile webui work properly again

**Comment by fichtner** [comment]
Maybe if you give device information or any sort of direction where this happens that would help?

**Comment by MainServerSpace** [comment]
I gave all the details it’s happening on my iOS device using any browser I’m currently using the newest version and all the steps to reproduce is to login to mobile and go the the firmware and check area and perform a security audit or health audit and then the buttons become stuck and don’t allow for the security or health audit to be performed. I shared a video of it happening but can’t really see the clicks

**Comment by fichtner** [comment]
> I gave all the details

Ok, but...

> it’s happening on my iOS device using any browser

That's something you haven't said before. ;)

I've tried iOS and the layout is a bit broken but the buttons do work... layout issues possibly since iOS 16, but you haven't said that either ;)


Cheers,
Franco

**Comment by MainServerSpace** [comment]
You do have a point but this was working previously but since router upgrade its been doing this. I did put a broad term of Mobile devices in my first comment message in light of others and not just my self.

**Comment by fichtner** [comment]
We removed Internet Explorer support in 22.7 but I doubt that iOS would complain about that specifically. Problem with phones is that getting to the JS error is tedious... if the button doesn't work the whole page may be broken. The check for updates button works or not?

**Comment by fichtner** [comment]
BTW, can you try the standard theme? What is odd to me is that the audit button has a primary color but it's not a primary button...

**Comment by MainServerSpace** [comment]
> We removed Internet Explorer support in 22.7 but I doubt that iOS would complain about that specifically. Problem with phones is that getting to the JS error is tedious... if the button doesn't work the whole page may be broken. The check for updates button works or not?

Yes the check for updates for OS work the others don't

**Comment by MainServerSpace** [comment]
> BTW, can you try the standard theme? What is odd to me is that the audit button has a primary color but it's not a primary button...

Can try that using the custom theme

**Comment by fichtner** [comment]
No I'm not a maintainer of plugin themes and it would help for you to confirm where the issue actually lies. It just indicates JS is still working fine then.

**Comment by MainServerSpace** [comment]
![FAAA9AFB-8CBD-4144-8608-DF6A2CA9BD49](https://user-images.githubusercontent.com/86547540/208418596-c40ee8ce-ef6b-4840-8317-59b3f5243097.png)

regular theme doesn’t work either it actually cuts off the drop down box

**Comment by MainServerSpace** [comment]
You can partially see it but can’t scroll or anything like that

**Comment by fichtner** [comment]
Fair enough, that's the issue I'm seeing. But I still suspect that's first and foremost iOS screwing it up. I can't look into this in any (CSS) depth today.

**Comment by MainServerSpace** [comment]
Fair enough just handy as I'm more mobile and on my phone with my network. Don’t push your guy’s self to hard just a cleanup. I appreciate it none the less.

**Comment by kulikov-a** [comment]
Hi. it looks like old Dropdown inside Table Responsive vs. iOS issue (https://github.com/twbs/bootstrap/issues/29313)
fastest way is to remove .table-responsive from Status table imho
@MainServerSpace can you test how it looks with `opnsense-patch -a kulikov-a ea44ebc` ?

**Comment by MainServerSpace** [comment]
Yes I can just waking back up like 30 minutes or so I can provide output

**Comment by MainServerSpace** [comment]
![6BEF7517-04C1-479A-91B3-6C0BC6336CBF](https://user-images.githubusercontent.com/86547540/208528386-fa7b7caa-f3d6-4241-a728-55eb04ca5eaf.png)

it works you hit it right on the nail

**Comment by MainServerSpace** [comment]
![767864CA-40A7-4298-8EDA-C1351F46F119](https://user-images.githubusercontent.com/86547540/208528663-cf23a66d-78c7-4549-93c6-37d6ea993fc5.png)

**Comment by fichtner** [comment]
@kulikov-a open a PR when you can. I don't want to interfere. :)

**Comment by kulikov-a** [comment]
@MainServerSpace thanks!
@fichtner @MainServerSpace give me a day please (a bit busy right now). knowing what the issue is, now i want to make this page responsive )

**Comment by MainServerSpace** [comment]
@kulikov-a your welcome and for sure I feel that been working a bunch of long hours as well. I appreciate all your help with fixing this

**Comment by kulikov-a** [comment]
@MainServerSpace can you try how it works and looks with  `opnsense-patch -a kulikov-a ea44ebc fd3bbed` please?
this will reverse previouse and apply new version. have no actual android device so was rely on chrome\ff dev console: after moving `.table-responsive` in the right place tables starts to look nicer but all mobile devices loses dropdowns. patch expands them upword on mobile devices and restores table responsivness.


https://user-images.githubusercontent.com/36099472/208869579-c51eeafc-c6ac-4d88-865f-aa2d297104c4.mp4

**Comment by MainServerSpace** [comment]
Yes I will test it shortly and let you know if it still works

**Comment by MainServerSpace** [comment]
![36290BE9-7EC0-4AF4-BE95-95729A8AFA7A](https://user-images.githubusercontent.com/86547540/208891758-e1c902af-7715-4648-91f8-223dfa126a0b.png)

still works needs some improvements tho from the last time and that is the box should drop down and not drop up and I think the buttons overall could be pushed over a hair

**Comment by MainServerSpace** [comment]
Could live with the drop up but I could see that becoming a issue down the line especially if you are editing something and need to use the drop down box as a point of reference particularly my VPN setup.

**Comment by MainServerSpace** [comment]
Personally I think buttons should sit flush with the checked on minimizing unneeded overhead and could make the page look more crispy

**Comment by MainServerSpace** [comment]
Thats understandable like I said would be nice to have them drop down than up that would fix the table issue as a whole. We definitely are on to something with which what has been done so far.

**Comment by kulikov-a** [comment]
@MainServerSpace with `opnsense-patch -a kulikov-a 8d52743`  (above fd3bbed. no need to revert) buttons should stay in the same line at all times
![tempsnip](https://user-images.githubusercontent.com/36099472/208927441-17eac326-42fe-4767-90a8-afa3ea08db6a.png)

that is, the page is truly responsive now, the buttons do not move apart, the options are visible..
I'd stop there if @fichtner doesn't mind

> I could see that becoming a issue down the line especially if you are editing something and need to use the drop down box as a point of reference particularly my VPN setup

I searched through the code and did not find any places where there issues could still be (and the patch only applies to the firmware page). please point out where else there might be problems?
and i agree with @fichtner that using portrait mode is akin to masochism (portrait mode is shown in the video to better demonstrate the responsiveness of tabs) :)

> would be nice to have them drop down

afaiu this will require some style hacking (actually the dropdowns vs responsiveness compatibility issue is even pointed out in the bootstrap docs) that won't outlive a possible migration to bootstrap 4 or 5 (does it make sense to do a temporary workaround?) OR require to change the location of the buttons (take them out of the table or move them up the table), which will violate the general style

**Comment by MainServerSpace** [comment]
Well that first fix you did, did fix it so not sure how it would breakSent from my iPhoneOn Dec 21, 2022, at 6:35 AM, kulikov-a ***@***.***> wrote:﻿
@MainServerSpace with opnsense-patch -a kulikov-a 8d52743  (above fd3bbed. no need to revert) buttons should stay in the same line at all times

that is, the page is truly responsive now, the buttons do not move apart, the options are visible..
I'd stop there if @fichtner doesn't mind

I could see that becoming a issue down the line especially if you are editing something and need to use the drop down box as a point of reference particularly my VPN setup

I searched through the code and did not find any places where there issues could still be (and the patch only applies to the firmware page). please point out where else there might be problems?
and i agree with @fichtner that using portrait mode is akin to masochism (portrait mode is shown in the video to better demonstrate the responsiveness of tabs) :)

would be nice to have them drop down

afaiu this will require some style hacking (actually the dropdowns vs responsiveness compatibility issue is even pointed out in the bootstrap docs) that won't outlive a possible migration to bootstrap 4 or 5 (does it make sense to do a temporary workaround?) OR require to change the location of the buttons (take them out of the table or move them up the table), which will violate the general style

**Comment by MainServerSpace** [comment]
@kulikov-a this was fine previously before 22.1.9 I think the last stable stable for the table was 22.1.7 or earlier

**Comment by kulikov-a** [comment]
@MainServerSpace 

> this was fine previously before 22.1.9 I think

oh, not ready to look for an even older version)
tbh, I still don’t understand why you insist on dropdowns but.. can you try `opnsense-patch -a kulikov-a 9d34042` then (need to revert all previous patches first) please?
its css/html issue workaround that should survive on possible bootstrap upgrade.
so it fully responsive page, no menu cuts, buttons do not move apart

**Comment by MainServerSpace** [comment]
I use OPNSense to serve VPN Tunnels on the fly. So having a way to check my health and security every now and then is key to find new vulnerabilities that are the software so I can be proactive with my network. I will test and report back.

**Comment by MainServerSpace** [comment]
@kulikov-a works as intended. Can provide screenshots if needed for reference

**Comment by MainServerSpace** [comment]
@kulikov-a only thing I see which is total fine because data can be seen as needed but that is IPSec Tunnel UI doesn't respect the other stuff.

**Comment by MainServerSpace** [comment]
![2F05EA6E-0CBB-485A-8FC7-536BAFCB6803](https://user-images.githubusercontent.com/86547540/209102540-7459751f-aabb-4646-9586-36aac3956e79.jpeg)

overhangs but honestly like it more this way because I can go page by page and can see stuff in full without issue. Looks and feels more like the computer browser edition on mobile without requesting desktop browser

**Comment by MainServerSpace** [comment]
I see one or two regressions with this patch but not huge regarding the page lineup. This new patch respects all pages but not IPSEC and VPN Status Page and a few others. @kulikov-a I think we should push this out. Out of respect of others including myself. This might just have to be a task I can focus on in my free time. Seems we fixed the dropdown but now pages are hanging off the side.
![download](https://user-images.githubusercontent.com/86547540/209316444-97556d75-002f-460c-a66c-fd8595e6c105.jpeg)

**Comment by kulikov-a** [comment]
@MainServerSpace the patch is only related to the firmware page. the table on the _nat_1to1 page just wasn't made .responsive from the start: https://github.com/opnsense/core/blob/d7be43f91329cf2b3f389e7c3421e149efa3aaa8/src/www/firewall_nat_1to1.php#L273
if you suggest more similar places, maybe I will make a separate request for them?

**Comment by MainServerSpace** [comment]
@kulikov-a 
![0A0D04C4-8063-45A1-9D45-B976992CC204](https://user-images.githubusercontent.com/86547540/209335606-8b8775af-704f-4108-b3bf-df0e72e42a1c.png)
Without the patch installed vs the previous image with the patch installed. Without patch it respects the proper design. Same area as well. I can respectfully grab each section that needs that with the new patch if that is needed but as far as I seen it the other stuff was good before patch was applied

**Comment by kulikov-a** [comment]
please compare with the same theme (opnsense)

**Comment by MainServerSpace** [comment]
@kulikov-a yeah I see that. So what I will do is when I wake up if needed I can reply back here and give a full response to which pages etc could use a bit of updating so we can push separate PRs for this

**Comment by MainServerSpace** [comment]
![Uploading 8D985285-D00B-4636-92F5-9511D6C12B87.png…]()

**Comment by MainServerSpace** [comment]
@kulikov-a just getting done with the holidays will reply this week sorry for the delay

**Comment by MainServerSpace** [comment]
@kulikov-a so basic places really I would like to see fixed from a mobile aspect is mainly the firewall rules, crowdsec under services, the vpn page, and the plugin and status check which is done. Thats about all I use and see could be fixing. Like mentioned its a bigger issue but if we was to go on there are other pages.

**Comment by kulikov-a** [comment]
@MainServerSpace Hi! I will try to look at the pages mentioned (it would be great if you provide relative links so that we're talking about the same things). but only after already opened tickets (it seems that I have been too much lately)

**Comment by MainServerSpace** [comment]
For sure. Yeah plan was to do it in separate halfs because to be frankly honest @kulikov-a this seems like a overall bigger issue so to separate everything and keep track of each pr made would probably be better if something down the line as to change etc then it seems logical to do this in steps

**Comment by MainServerSpace** [comment]
@kulikov-a like @fichtner mentioned is he is probably doesn't want to do this all up front with things so I think like mentioned and the most logical approach would for us to do this as individual PRs and track stuff as it goes. Seems the check for update and health security is work as intended from my testing so lets close this PR after it hits upstream to ensure it works all across and that it doesn't cause complaints and then if it doesn't I will push my other PRs with intended links for update

**Comment by kulikov-a** [comment]
@MainServerSpace Great! sounds logical

**Comment by MainServerSpace** [comment]
For sure appreciate your help @kulikov-a

**Comment by fichtner** [comment]
@kulikov-a not sure if this was intentional or not but in any case consider me sceptical of the split view here:

<img width="1448" alt="Screenshot 2023-01-19 at 12 20 12" src="https://user-images.githubusercontent.com/1915288/213429841-0606796e-d4db-4743-8095-ca63990dc320.png">

It took me a while to correlate this oddity with the commit :)

**Comment by kulikov-a** [comment]
@fichtner hm. update container not hiding after check? 
weird and definitely not intentional
I'm looking for a reason..

**Comment by fichtner** [comment]
To be precise the log container is not hiding as it should. The split view has some benefits but I doubt it looks cleaner. :)

**Comment by kulikov-a** [comment]
aargh..stupid typo at https://github.com/opnsense/core/blob/a1569e8938370c766420739e29b87bc218770576/src/opnsense/mvc/app/views/OPNsense/Core/firmware.volt#L828 
so sorry
will check and add pr in a minute

**Comment by fichtner** [comment]
hehe, new id :) no rush, thanks for looking into it
